apollo-router/src/plugins/telemetry/fmt_layer.rs

```diff
@@ -15,37 +28,198 @@ pub(crate) fn create_fmt_layer(
+        let span = ctx.span(id).expect("Span not found, this is a bug");
+        let mut extensions = span.extensions_mut();
+        if let Some(fields) = extensions.get_mut::<LogAttributes>() {
+            let mut visitor = FieldsVisitor::default();
+            values.record(&mut visitor);
+            fields.extend(
+                visitor
+                    .values
+                    .into_iter()
+                    .filter_map(|(k, v)| Some((k.into(), v.maybe_to_otel_value()?))),
+            );
+        } else {
+            eprintln!("cannot access to LogAttributes, this is a bug");
+        }
+    }
+
+    fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
+        thread_local! {
+            static BUF: RefCell<String> = RefCell::new(String::new());
+        }
+
+        BUF.with(|buf| {
+            let borrow = buf.try_borrow_mut();
+            let mut a;
+            let mut b;
+            let mut buf = match borrow {
+                Ok(buf) => {
+                    a = buf;
+                    &mut *a
+                }
+                _ => {
+                    b = String::new();
+                    &mut b
+                }
+            };
+            if self.fmt_event.format_event(&ctx, &mut buf, event).is_ok() {
+                let mut writer = std::io::stdout();
+                if let Err(err) = std::io::Write::write_all(&mut writer, buf.as_bytes()) {
+                    eprintln!("cannot flush the logging buffer, this is a bug: {err:?}");
+                }
+            }
+            buf.clear();
+        });
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct FieldsVisitor<'a> {
+    pub(crate) values: HashMap<&'a str, serde_json::Value>,
+}
+
+impl<'a> Default for FieldsVisitor<'a> {
+    fn default() -> Self {
+        Self {
+            values: HashMap::with_capacity(0),
+        }
+    }
+}
+
+impl<'a> field::Visit for FieldsVisitor<'a> {
+    /// Visit a double precision floating point value.
+    fn record_f64(&mut self, field: &Field, value: f64) {
+        self.values
+            .insert(field.name(), serde_json::Value::from(value));
+    }
+
+    /// Visit a signed 64-bit integer value.
+    fn record_i64(&mut self, field: &Field, value: i64) {
+        self.values
+            .insert(field.name(), serde_json::Value::from(value));
+    }
+
+    /// Visit an unsigned 64-bit integer value.
+    fn record_u64(&mut self, field: &Field, value: u64) {
+        self.values
+            .insert(field.name(), serde_json::Value::from(value));
+    }
+
+    /// Visit a boolean value.
+    fn record_bool(&mut self, field: &Field, value: bool) {
+        self.values
+            .insert(field.name(), serde_json::Value::from(value));
+    }
+
+    /// Visit a string value.
+    fn record_str(&mut self, field: &Field, value: &str) {
+        let field_name = field.name();
+        match field_name {
+            "code.filepath" | "code.namespace" | "code.lineno" | "thread.id" | "thread.name" => {}
```

Comment:
does this translate to `field_name == "code.filepath" || field_name == "code.namespace" || etc`? We had an issue elsewhere where we checked if a span name was in a list, and that was surprisingly expensive. It was fixed by moving to a `HashSet`: https://github.com/apollographql/router/pull/3656/